plutus-use-cases/src/Plutus/Contracts/Auction.hs

```diff
@@ -164,17 +191,34 @@ data AuctionLog =
     deriving stock (Show, Generic)
     deriving anyclass (ToJSON, FromJSON)
 
+data AuctionError =
+    StateMachineContractError SM.SMContractError -- ^ State machine operation failed
+    | ThreadTokenError Currency.CurrencyError -- ^ Thread token could not be created
+    | AuctionContractError ContractError -- ^ Endpoint, coin selection, etc. failed
+    deriving stock (Haskell.Eq, Haskell.Show, Generic)
+    deriving anyclass (ToJSON, FromJSON)
+
+makeClassyPrisms ''AuctionError
+
+instance AsContractError AuctionError where
+    _ContractError = _AuctionContractError . _ContractError
+
+instance SM.AsSMContractError AuctionError where
+    _SMContractError = _StateMachineContractError . SM._SMContractError
 
 -- | Client code for the seller
-auctionSeller :: Value -> Slot -> Contract (Maybe (Last AuctionState)) SellerSchema SM.SMContractError ()
+auctionSeller :: Value -> Slot -> Contract AuctionOutput SellerSchema AuctionError ()
 auctionSeller value slot = do
+    threadToken <- mapError ThreadTokenError Currency.createThreadToken
```

Comment:
Should we not also burn it at the end? It would be nice to pull the forging/burning into the state machine operation so this stuff was automatic, but I get that this is the MVP.

Reply:
Yeah I'll have a look to see how hard it actually is.